aws-datastore/src/main/java/com/amplifyframework/datastore/appsync/AppSyncRequestFactory.java

```diff
@@ -108,7 +108,16 @@ private AppSyncRequestFactory() {}
             }
             if (!QueryPredicates.all().equals(predicate)) {
                 String filterType = "Model" + Casing.capitalizeFirst(modelSchema.getName()) + "FilterInput";
-                builder.variable("filter", filterType, parsePredicate(predicate));
+                QueryPredicate syncPredicate = predicate;
+                if (syncPredicate instanceof QueryPredicateOperation) {
+                    // When a filter is provided, wrap it with a predicate group of type AND.  By doing this, it enables
+                    // AppSync to optimize the request by performing a DynamoDB query instead of a scan.  If the
+                    // provided syncPredicate is already a QueryPredicateGroup, this is not needed.  If the provided
+                    // group is of type AND, the optimization will occur.  If the top level group is OR or NOT, the
+                    // optimization is not possible anyway.
+                    syncPredicate = new QueryPredicateGroup(QueryPredicateGroup.Type.AND, Arrays.asList(syncPredicate));
```

Comment:
This might look better wrapped into a factory method:

```java 
QueryPredicateGroup.andOf(syncPredicate)
```
It would prevent the access escalation of the constructor, too.

Reply:
I had considered something like that, but couldn't come up with a good name for the method itself.  I think `andOf` works though - I'll do that!